app/middleware/csrf.js

```diff
@@ -8,17 +8,17 @@ const logger = require('../utils/logger')(__filename)
 const { getLoggingFields } = require('../utils/logging_fields_helper')
 const session = require('../utils/session')
 const responseRouter = require('../utils/response_router')
-const chargeParam = require('../services/charge_param_retriever')
+const validateSessionCookie = require('../services/validate_session_cookie')
```

Comment:
I wonder if it would make sense for these actions to be grouped by the fact they all perform actions on "sessions" rather than being split across multiple areas as they currently are. 

Naming is always challenging in programming but frontend seems to have a folder called `services` and `utils` and uses the two interchangeably. 

The name `validate` also doesn't imply that a charge ID will be returned.

There already exists a `/utils/session.js`, I think it's quite clear to understand that these are a set of utilities that act on cookie sessions. What do you think of the following:

```js
const {  getChargeIdIfExists } = require('utils/session')

// returns chargeId if exists and a falsey value otherwise
const chargeId = getChargeIdIfExists(req)
```

Reply:
Maybe you are correct that the validation code can belongs in `/utils/session.js.` I think it eventually should return true/false (should only validate).  Then in retrieve_charge will only build the charge, shouldn't check session something like this.

`const chargeId = req.params.chargeId ? req.params.chargeId : req.body.chargeId
req.chargeId = chargeId
 Charge(req.headers[CORRELATION_HEADER]).find(chargeId, getLoggingFields(req))
      .then(data => {
        req.chargeData = data
        setLoggingField(req, GATEWAY_ACCOUNT_ID, data.gateway_account.gateway_account_id)
        setLoggingField(req, GATEWAY_ACCOUNT_TYPE, data.gateway_account.type)
        setLoggingField(req, PROVIDER, data.gateway_account.payment_provider)
        next()
   })
  .catch((err) => {
        responseRouter.systemErrorResponse(req, res, 'Error finding charge in middleware', withAnalyticsError(), err)
   })`

Then 'validate_session_cookie' would eventually look something like

`const {  validateSessionCookie } = require('utils/session')
const validateSessionCookie = validateSessionCookie(req)
if(!validateSessionCookie)){
    responseRouter.response(req, res, 'UNAUTHORISED', withAnalyticsError())
}
next() // not sure this is need
` 
Then in route we will have we have `...,validateSessionCookie, retrieveCharge, ....` or just `retrieveCharge.` let's discuss this at 11:00am

Reply:
Implemented this pattern